include/zephyr/dt-bindings/clock/stm32_common_clocks.h

```diff
@@ -23,30 +23,31 @@
 /** Helper macros to pack RCC clock source selection register info in the DT */
 #define STM32_DT_CLKSEL_REG_MASK	0xFFFFU
 #define STM32_DT_CLKSEL_REG_SHIFT	0U
-#define STM32_DT_CLKSEL_SHIFT_MASK	0x3FU
+#define STM32_DT_CLKSEL_SHIFT_MASK	0x1FU
 #define STM32_DT_CLKSEL_SHIFT_SHIFT	16U
-#define STM32_DT_CLKSEL_MASK_MASK	0x1FU
-#define STM32_DT_CLKSEL_MASK_SHIFT	22U
-#define STM32_DT_CLKSEL_VAL_MASK	0x1FU
-#define STM32_DT_CLKSEL_VAL_SHIFT	27U
+#define STM32_DT_CLKSEL_WIDTH_MASK	0x3U
+#define STM32_DT_CLKSEL_WIDTH_SHIFT	21U
+#define STM32_DT_CLKSEL_VAL_MASK	0xFFU
+#define STM32_DT_CLKSEL_VAL_SHIFT	24U
 
 /**
  * @brief Pack STM32 source clock selection RCC register bit fields for the DT
  *
  * @param val    Clock configuration field value
- * @param mask   Mask of register field in RCC register
- * @param shift  Position of field within RCC register (= field LSB's index)
+ * @param msb    Field MSB's index
+ * @param lsb    Field LSB's index
  * @param reg    Offset to target clock configuration register in RCC
  *
+ * @note Internally, the data are stored as follows
  * @note 'reg' range:    0x0~0xFFFF   [ 00 : 15 ]
- * @note 'shift' range:  0~63         [ 16 : 21 ]
- * @note 'mask' range:   0x00~0x1F    [ 22 : 26 ]
- * @note 'val' range:    0x00~0x1F    [ 27 : 31 ]
+ * @note 'shift' range:  0~31         [ 16 : 20 ]
+ * @note 'width' range:  0~7          [ 21 : 23 ]
```

Comment:
```suggestion
 * @note 'width' range:  0~7          [ 21 : 23 ]  Value encodes bit fields width minus 1
```
Would explicitly address extra comment from https://github.com/zephyrproject-rtos/zephyr/pull/97725/commits/e29ae439b835f26f1315533d1812f6ffb13670d6#r2451959811.